content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+        task = self.info["task"]
+        n_classes = len(self.info["label"])
+
+        self.context = context
+        if self.context.get_hparam("model_flag") == "resnet18":
+            model = resnet18(pretrained=False, num_classes=n_classes)
+        elif self.context.get_hparam("model_flag") == "resnet50":
+            model = resnet50(pretrained=False, num_classes=n_classes)
+        else:
+            raise NotImplementedError
+
+        self.model = self.context.wrap_model(model)
+
+        optimizer = torch.optim.Adam(
+            self.model.parameters(), lr=self.context.get_hparam("lr")
+        )
+        self.optimizer = self.context.wrap_optimizer(optimizer)
+
+        if self.context.get_hparam("task") == "multi-label, binary-class":
+            self.criterion = nn.BCEWithLogitsLoss()
+        else:
+            self.criterion = nn.CrossEntropyLoss()
+```
+
+#### Step 1.3 Port Data Loaders  
+
+Next, we’ll port the training and evaluation data loaders to the following class functions as follows. This is the code that trains one batch of data. You no longer need the standard for loop to iterate over batches or epochs inside these functions, since Determined will handle the training loop for you. 
+
+```python
+    def build_training_data_loader(self) -> DataLoader:
+        DataClass = getattr(medmnist, self.info["python_class"])
+
+        if self.context.get_hparam("resize"):
+            data_transform = transforms.Compose(
+                [
+                    transforms.Resize((224, 224), interpolation=PIL.Image.NEAREST),
+                    transforms.ToTensor(),
+                    transforms.Normalize(mean=[0.5], std=[0.5]),
+                ]
+            )
+        else:
+            data_transform = transforms.Compose(
+                [transforms.ToTensor(), transforms.Normalize(mean=[0.5], std=[0.5])]
+            )
+
+        train_dataset = DataClass(
+            split="train",
+            transform=data_transform,
+            download=False,
+            as_rgb=True,
+            root=DATASET_ROOT,
+        )
+        train_loader = determined.pytorch.DataLoader(
+            dataset=train_dataset,
+            batch_size=self.context.get_per_slot_batch_size(),
+            shuffle=True,
+        )
+
+        return train_loader
+```
+
+In each function, we initialize and return the relevant PyTorch DataLoader object (e.g. `val_loader from build_validation_data_loader`, and `train_loader in build_training_data_loader`).  
+
+#### Step 1.4 Port Training and Evaluation Functions 
+
+Finally, we’ll port our training and evaluation functions to the following class functions by including the relevant steps to train and evaluate our model on one batch of data. Here, make sure to remove the for-loop iterating over epochs and include only the relevant code corresponding to one data batch. HPE Machine Learning Development Environment handles the training loop behind the scenes.  
```

Comment:
Modify the first part of this to:
"Finally, port your training and evaluation functions.... to train and evaluate the model on one batch of data."